### DISCUSSION TITLE
Using Pytroch On AMD GPU2 AMD Radeon RX 6600M Discrete/Hybrid

### DISCUSSION THREAD

**Discussion by nikhil7975** [ROCm/ROCm#2266] [Q&A]
The reason why I am wiriting here  is because I have AMD ryzen 9 processor and AMD Raedon GPU. I wish to run pytorch, which is essential for deep learning models for my MSc project. When I searched the documentation, link: https://pytorch.org/get-started/locally/

<img width="652" alt="image" src="https://github.com/RadeonOpenCompute/ROCm/assets/110478152/96ae8ba6-5107-4548-9e37-b948036960d5">

RoCm is something that I am allowed download to use pytroch, I have downloaded this, unfortunately when I ask this in my code, torch.cuda.is_available(), it always returns false, that means it is not using my GPU even though I have GPU,

Regarding my GPU version please refer to the snippet, 

<img width="553" alt="image" src="https://github.com/RadeonOpenCompute/ROCm/assets/110478152/df4f55c4-dcb5-422e-b8bd-c8dd3e2fa03c">

As per the AMD GPU support list: https://docs.amd.com/en/docs-5.3.0/release/gpu_os_support.html

Pytorch is not using my GPU, even tensorflow is not using my GPU. How do I fix this? Please escalate the issue to the relevant team as this is important. Please note laptop is HP Omen.              

As per the ROCm documentation my GPU is supported for this. I am currently using  The Windows Subsystem for Linux (WSL), through which I was able to install Linux and which has allowed me to run visual studio in linux terminal. 

I would truly appreciate if someone can guide me here.

**Comment by briansp2020** [comment]
ROCm under WSL is not supported yet for any AMD hardware. AMD is working on releasing Windows support but nobody knows whether it is pure Windows or WSL as well.

**Reply by nikhil7975** [reply]
Thanks a ton for this information. That means, I should be able to use RocM in Linux system alone?

**Reply by nikhil7975** [reply]
Well apparently tried on Linux OS as well, it has nothing to do with the O.S, it has everything to do with the GPU support. For some reason my GPU is not supported for this. What a waste of purchasing this laptop.

**Reply by yk70840** [reply]
Hello so did you make it work, if yes can you please tell how ?

**Comment by zichguan-amd** [comment]
Hi @nikhil7975, unfortunately we don't have official support for mobile GPUs, but you can try the workaround here https://github.com/ROCm/ROCm/issues/1756. PyTorch is not supported on Windows right now, see https://rocm.docs.amd.com/projects/install-on-windows/en/develop/reference/component-support.html for ROCm component support on Windows. However, you can set up PyTorch on WSL following the installation guide here https://rocm.docs.amd.com/projects/radeon/en/latest/docs/install/wsl/install-pytorch.html
